gha: unify windows builds into one workflow and drop sha ID correlating

jsliacan opened the pull request:
Addresses Issue #3975 

The solution with many side benefits (see commit log) is to unify what was formerly two workflows `build-tests.yaml` and `windows-installer.yaml` into `windows-artifacts.yaml`. This workflow then triggers `windows-e2e.yaml` via `workflow_run` and therefore, `windows-e2e.yaml` has access to all artifacts from `windows-artifacts.yaml` directly - no need to correlate via commit sha or similar. 

As a side effect, `windows-e2e.yaml` only gets triggered *after* all artifacts are produced, so we can rearrange steps inside `windows-e2e.yaml` in a way that postpones target host provisioning until after we know we have everything else ready (no provisioning if some precondition fails). 

Finally, parametrize commit status context by matrix entries, to futureproof.

A clean run (with only a `config.feature` run) can be seen checked as part of this dummy PR https://github.com/jsliacan/crc/pull/9. Here's the actual action: https://github.com/jsliacan/crc/actions/runs/7668762265

anjannath commented on .github/workflows/build-tests.yml:
```diff
@@ -19,16 +19,25 @@ jobs:
       
+      - name: Work out correct id
+        run: |
+          COMMIT_ID=${{ toJSON(github) }} | jq -r '.event.after')
```
an opening `(` is missing here

jsliacan replied:
Yes, thanks :)

Commit:
gha: unify all windows artifacts builds into one workflow

This helps circumvent multiple complications. One, windows-e2e
workflow does not need to wait for 2 workflows to finish (which is
hard for a workflow_run triggered workflow). Two, we do not need to
worry about matching the right artifacts from builder workflows with
the right workflow_run that needs them (formerly used commit sha to do
this). Third, the organization of workflows becomes logical: builder
that prepares artifacts, and tester that consumes them and runs tests.

Commit:
gha: clarify information in the commit status

Parametrizing by workflow's matrix entries future-proofs the workflow.

Commit:
gha: rearrange steps to avoid provisioning unnecessarily

Move all steps that might fail to before the provisioning step to
avoid creating a machine if simple steps needed to run tests on it
fail anyways.
